services/httpimporter/service/service.go

```diff
@@ -61,24 +52,15 @@ func (h *HTTPImporter) httpPostUplink() http.HandlerFunc {
 			return
 		}
 
-		pipelineModel, err := h.pipeline.Get(pipelineID.String())
-		if err != nil {
-			web.HTTPError(rw, err)
-			return
-		}
-		steps := pipelineModel.Steps
-
-		msg := pipeline.NewMessage(pipelineID.String(), steps)
-		msg.Payload = payload
+		// TODO: Implement authentication logic then add owner to ingress
+		var ownerID int64 = 1
 
-		if err := h.queue.Publish(msg); err != nil {
-			web.HTTPError(rw, err)
-			return
-		}
+		dto := processing.CreateIngressDTO(pipelineID, ownerID, payload)
+		h.publisher <- dto
 
 		web.HTTPResponse(rw, http.StatusAccepted, &web.APIResponseAny{
 			Message: "Received uplink message",
-			Data:    msg.ID,
+			Data:    dto.TracingID.String(),
```

Comment:
can TracingID be nil?

Reply:
See previous comment about processing.CreateIngressDTO